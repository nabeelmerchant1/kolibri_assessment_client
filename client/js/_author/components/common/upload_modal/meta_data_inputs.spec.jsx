import React            from 'react';
import { shallow }      from 'enzyme';
import MetaDataInputs   from './meta_data_inputs';
import languages        from '../../../../constants/language_types';

describe('meta data inputs component', () => {
  let props;
  let result;
  let calledUpdate;

  beforeEach(() => {
    calledUpdate = false;
    props = {
      selectedLanguage: languages.languageTypeId.english,
      labelName: () => {},
      metadataTypes: ['altText'],
      metaData: {
        language: 'English',
      },
      updateMetadata: () => { calledUpdate = true; },
    };
    result = shallow(<MetaDataInputs {...props} />);
  });

  it('renders a snapshot of the component', () => {
    expect(result).toMatchSnapshot();
  });

  it('calls updateMetadata function', () => {
    expect(calledUpdate).toBeFalsy();
    result.find('.au-c-text-input').simulate('change', { target: { value: 'something new' } });
    expect(calledUpdate).toBeTruthy();
  });
});
