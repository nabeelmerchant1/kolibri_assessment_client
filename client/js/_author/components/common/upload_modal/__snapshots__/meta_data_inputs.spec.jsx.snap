// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`meta data inputs component renders a snapshot of the component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
        className="au-c-input au-c-input-label--left"
    >
        <label
            htmlFor="meta_upload_altText"
        />
        <div
            className="au-c-input__contain"
        >
            <input
                className="au-c-text-input au-c-text-input--smaller"
                id="meta_upload_altText"
                onChange={[Function]}
                tabIndex="0"
                type="text"
                value=""
            />
            <div
                className="au-c-input__bottom"
            />
        </div>
    </div>
</div>,
  "nodes": Array [
    <div>
      <div
            className="au-c-input au-c-input-label--left"
      >
            <label
                  htmlFor="meta_upload_altText"
            />
            <div
                  className="au-c-input__contain"
            >
                  <input
                        className="au-c-text-input au-c-text-input--smaller"
                        id="meta_upload_altText"
                        onChange={[Function]}
                        tabIndex="0"
                        type="text"
                        value=""
                  />
                  <div
                        className="au-c-input__bottom"
                  />
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <metaDataInputs
        labelName={[Function]}
        metaData={
                Object {
                        "language": "English",
                      }
        }
        metadataTypes={
                Array [
                        "altText",
                      ]
        }
        selectedLanguage="639-2%3AENG%40ISO"
        updateMetadata={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "labelName": [Function],
          "metaData": Object {
            "language": "English",
          },
          "metadataTypes": Array [
            "altText",
          ],
          "selectedLanguage": "639-2%3AENG%40ISO",
          "updateMetadata": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
                    className="au-c-input au-c-input-label--left"
          >
                    <label
                              htmlFor="meta_upload_altText"
                    />
                    <div
                              className="au-c-input__contain"
                    >
                              <input
                                        className="au-c-text-input au-c-text-input--smaller"
                                        id="meta_upload_altText"
                                        onChange={[Function]}
                                        tabIndex="0"
                                        type="text"
                                        value=""
                              />
                              <div
                                        className="au-c-input__bottom"
                              />
                    </div>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <div
                    className="au-c-input au-c-input-label--left"
          >
                    <label
                              htmlFor="meta_upload_altText"
                    />
                    <div
                              className="au-c-input__contain"
                    >
                              <input
                                        className="au-c-text-input au-c-text-input--smaller"
                                        id="meta_upload_altText"
                                        onChange={[Function]}
                                        tabIndex="0"
                                        type="text"
                                        value=""
                              />
                              <div
                                        className="au-c-input__bottom"
                              />
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <metaDataInputs
    labelName={[Function]}
    metaData={
        Object {
            "language": "English",
          }
    }
    metadataTypes={
        Array [
            "altText",
          ]
    }
    selectedLanguage="639-2%3AENG%40ISO"
    updateMetadata={[Function]}
/>,
}
`;
